src/libraries/System.Data.OleDb/src/System.Data.OleDb.csproj

```diff
@@ -99,6 +99,21 @@
     <Compile Include="System\Data\ProviderBase\DbReferenceCollection.cs" />
     <Compile Include="System\Data\ProviderBase\WrappedIUnknown.cs" />
   </ItemGroup>
+  <ItemGroup Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net5.0')) and '$(TargetsWindows)' == 'true'">
+    <Compile Include="UnsafeNativeMethods.COMWrappers.cs" />
+    <Compile Include="OleDbComWrappers.cs" />
+    <Compile Include="DbPropSet.COMWrappers.cs" />
+    <Compile Include="OleDb_Util.COMWrappers.cs" />
+    <Compile Include="OleDbConnection.COMWrappers.cs" />
+    <Compile Include="OleDbException.COMWrappers.cs" />
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETStandard'">
```

Comment:
Is this right? Don't you want the `!` of the above condition on line 102?

Reply:
This would include net461-windows which is wrong as that's facade assembly. The current condition is probably missing a `'$(TargetsWindows)' == 'true'` condition.

Reply:
I guess I would use 

```xml
<ItemGroup Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net5.0')) and'$(IsPartialFacadeAssembly)' != 'true' and '$(TargetsWindows)' == 'true'">
```

to be symmetrical with the first ItemGroup condition above.

https://github.com/dotnet/runtime/blob/f6cc321fbd49b097364951137d3fd50c3b4872d2/src/libraries/System.Data.OleDb/src/System.Data.OleDb.csproj#L22

Reply:
@ViktorHofer and @eerhardt I made changes, but I'm not sure if this is proper way. All I can say, not it does not fails during build.

Reply:
Looks pretty good. My only comments would be to make the ItemGroup conditions in the same order, to help with readability:

```xml
  <ItemGroup Condition="'$(IsPartialFacadeAssembly)' != 'true' and '$(TargetsWindows)' == 'true'">
...
  <ItemGroup Condition="'$(IsPartialFacadeAssembly)' != 'true' and $([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', $(NetCoreAppCurrent))) and '$(TargetsWindows)' == 'true'">
...
  <ItemGroup Condition="'$(IsPartialFacadeAssembly)' != 'true' and !$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', $(NetCoreAppCurrent))) and '$(TargetsWindows)' == 'true'">
```

would become

```xml
  <ItemGroup Condition="'$(IsPartialFacadeAssembly)' != 'true' and '$(TargetsWindows)' == 'true'">
...
  <ItemGroup Condition="'$(IsPartialFacadeAssembly)' != 'true' and '$(TargetsWindows)' == 'true' and $([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', $(NetCoreAppCurrent)))">
...
  <ItemGroup Condition="'$(IsPartialFacadeAssembly)' != 'true' and '$(TargetsWindows)' == 'true' and !$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', $(NetCoreAppCurrent)))">
```

And then change the `IsTargetFrameworkCompatible` to use `net5.0`, since COMWrappers were introduced in `net5.0`. This helps with maintainability in the long term, when we update `NetCoreAppCurrent` to be `net7.0` in the future.